starters/cms/src/data.ts

```diff
@@ -0,0 +1,182 @@
+import type { CollectionItemData, ManagedCollection, ManagedCollectionField } from "framer-plugin"
+
+import { framer } from "framer-plugin"
+import { PLUGIN_KEYS } from "./constants"
+
+export type ManagedCollectionFieldType = ManagedCollectionField["type"]
+
+export interface DataSource {
+    id: string
+    fields: readonly ManagedCollectionField[]
+    items: Record<string, unknown>[]
+}
+
+export function getDataSourceIds(): readonly string[] {
+    return ["articles", "categories"]
+}
+
+export async function getDataSource(dataSourceId: string, abortSignal?: AbortSignal): Promise<DataSource> {
+    const dataSource: {
+        id: string
+        fields: {
+            name: string
+            type: string
+        }[]
+        items: Record<string, unknown>[]
+    } = await fetch(`/datasources/${dataSourceId}.json`, { signal: abortSignal }).then(response => response.json())
```

Comment:
nit: lets split this on a few lines, its a bit weird to deconstruct directly from the fetch like that on the same line I think

Reply:
I also think keeping the DataSource type separate is easier to read rather than inline like this